### PR TITLE
fix(page): RouteFromHarAsync embed files

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-dotnet/issues/2811

### CODE REVIEW

File: src/Playwright/Core/Page.cs [hunk 1]
```diff
@@ -1357,6 +1357,9 @@ await Context.RecordIntoHarAsync(har, this, new()
                 Url = options?.Url,
                 UrlString = options?.UrlString,
                 UrlRegex = options?.UrlRegex,
+                Update = options.Update,
+                UpdateContent = options.UpdateContent,
+                UpdateMode = options.UpdateMode,
```
Comment (campersau) [inline src/Playwright/Core/Page.cs:1362]: `options` can be `null`

```suggestion
                Update = options?.Update,
                UpdateContent = options?.UpdateContent,
                UpdateMode = options?.UpdateMode,
```
(not sure if that compiles)

File: src/Playwright/Core/Page.cs [hunk 2]
```diff
@@ -1357,6 +1357,9 @@ await Context.RecordIntoHarAsync(har, this, new()
                 Url = options?.Url,
                 UrlString = options?.UrlString,
                 UrlRegex = options?.UrlRegex,
+                Update = options.Update,
+                UpdateContent = options.UpdateContent,
```
Comment (yury-s) [inline src/Playwright/Core/Page.cs:1361]:
```suggestion
                UpdateContent = options?.UpdateContent,
```
here and below?

**Review by yury-s — APPROVED** [review]

File: src/Playwright/Core/Page.cs (hunk 2 above)
Comment (campersau) [reply src/Playwright/Core/Page.cs:1361]: I fell for it as well but above is already a `options?.Update == true` check so `options` can't be `null` here.
Still might be good to make it consistent.

Comment (yury-s) [reply src/Playwright/Core/Page.cs:1361]: Agreed. @mxschmitt let's make it consistent.

Comment (mxschmitt) [reply src/Playwright/Core/Page.cs:1361]: good point! addressed.

**Pull request merged**